File: core/src/com/bosowski/oslark/screens/LoginScreen.java
package com.bosowski.oslark.screens;

import com.badlogic.gdx.Game;
import com.badlogic.gdx.Gdx;
import com.badlogic.gdx.Screen;
import com.badlogic.gdx.graphics.GL20;
import com.badlogic.gdx.scenes.scene2d.InputEvent;
import com.badlogic.gdx.scenes.scene2d.Stage;
import com.badlogic.gdx.scenes.scene2d.ui.Skin;
import com.badlogic.gdx.scenes.scene2d.ui.TextButton;
import com.badlogic.gdx.scenes.scene2d.ui.TextField;
import com.badlogic.gdx.scenes.scene2d.utils.ClickListener;
import com.bosowski.oslark.playerDomains.User;
import com.bosowski.oslark.main.Session;

import java.io.IOException;


public class LoginScreen extends AbstractGameScreen{

    Stage stage;

    TextField username;
    TextField password;
    TextButton login;


    public LoginScreen(Game game) {
        super(game);
        stage = new Stage();
        Skin fieldSkins = new Skin(Gdx.files.internal("uiskin.json"));
        username = new TextField("username", fieldSkins);
        password = new TextField("password", fieldSkins);
        login = new TextButton("Login", fieldSkins);
        username.setPosition(Gdx.graphics.getWidth()/2, Gdx.graphics.getHeight()/2);
        password.setPosition(username.getX(), username.getY()-50);
        login.setPosition(password.getX(), password.getY()-50);
        stage.addActor(username);
        stage.addActor(password);
        stage.addActor(login);
        login.addListener(new ClickListener(){
            @Override
            public boolean touchDown(InputEvent event, float x, float y, int point, int button){
                System.out.println("CLICKING");
                try {
                    Session.instance.login(username.getText(), password.getText());
                    String userJson = Session.instance.loadUser();
                    game.setScreen(new CharacterSelectionScreen(game, userJson));
                } catch (IOException e) {
                    e.printStackTrace();
                }
                return true;
            }
        });
    }



    @Override
    public void render(float deltaTime) {
        // Clear the buffer
        Gdx.gl.glClearColor(0, 0, 0, 1);
        Gdx.gl.glClear(GL20.GL_COLOR_BUFFER_BIT);

        stage.act(deltaTime);
        stage.draw();
    }

    @Override
    public void resize(int width, int height) {

    }

    @Override
    public void show() {
        Gdx.input.setInputProcessor(stage);
    }

    @Override
    public void hide() {

    }

    @Override
    public void pause() {

    }
}

File: OslarkGame/core/src/com/bosowski/oslark/screens/LoginScreen.kt
package com.bosowski.oslark.screens

import com.badlogic.gdx.Game
import com.badlogic.gdx.Gdx
import com.badlogic.gdx.graphics.GL20
import com.badlogic.gdx.scenes.scene2d.InputEvent
import com.badlogic.gdx.scenes.scene2d.ui.*
import com.badlogic.gdx.scenes.scene2d.utils.ClickListener
import com.bosowski.oslark.Assets
import com.bosowski.oslark.World
import com.bosowski.oslark.managers.NetworkManager
import java.io.IOException
import java.lang.Exception


class LoginScreen(game: Game) : AbstractGameScreen(game) {

  init {
    setUpUI()
  }

  override fun render(deltaTime: Float) {
    Gdx.gl.glClearColor(49f/255, 38f/255, 45f/255, 1f)
    Gdx.gl.glClear(GL20.GL_COLOR_BUFFER_BIT)

    stage.act(deltaTime)
    stage.draw()
  }

  override fun show() {
    Gdx.input.inputProcessor = stage
  }

  override fun hide() {

  }

  override fun pause() {

  }

  override fun setUpUI() {
    val backgroundContainer = Container<Image>()
    stage.addActor(backgroundContainer)
    backgroundContainer.setFillParent(true)
    val background = Image(Assets.textures["background"])
    backgroundContainer.actor = background
    stage.addActor(backgroundContainer)

    val mainTable = Table()
    stage.addActor(mainTable)
    mainTable.setFillParent(true)
    mainTable.debug = false

    val usernameGroup = Table()
    val usernameLabel = Label("Username:", fieldSkins)
    val usernameField = TextField("admin", fieldSkins)
    //usernameGroup.setPosition((Gdx.graphics.width / 2).toFloat(), (Gdx.graphics.height / 2).toFloat())
    usernameGroup.add(usernameLabel).width(100f)
    usernameGroup.add(usernameField)
    mainTable.add(usernameGroup).padBottom(15f)

    mainTable.row()
    val passwordGroup = Table()
    val passwordLabel = Label("Password:", fieldSkins)
    val passwordField = TextField("admin", fieldSkins)
    passwordField.isPasswordMode = true
    passwordField.setPasswordCharacter('*')
    //usernameGroup.setPosition((Gdx.graphics.width / 2).toFloat(), (Gdx.graphics.height / 2).toFloat())
    passwordGroup.add(passwordLabel).width(100f)
    passwordGroup.add(passwordField)
    mainTable.add(passwordGroup).padBottom(25f)

    mainTable.row()
    val loginButton = TextButton("Login", fieldSkins)
    mainTable.add(loginButton).width(100f)

    loginButton.addListener(object : ClickListener() {
      override fun touchDown(event: InputEvent?, x: Float, y: Float, point: Int, button: Int): Boolean {
        try {
          println("login result")
          if(NetworkManager.instance.login(usernameField.text, passwordField.text).contains("<title>        Login    </title>")){
            throw Exception("Invalid Credentials")
          }
          game.screen = CharacterSelectionScreen(game)
        } catch (e: IOException) {
          //todo: Display some message to user indicating why login was not successful.
          e.printStackTrace()
        } catch(e: Exception){
          e.printStackTrace()
        }
        return true
      }
    })

  }
}
